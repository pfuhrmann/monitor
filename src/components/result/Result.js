import React from 'react';
import PropTypes from 'prop-types';
import dayjs from 'dayjs';

import './Result.scss';
import TextSubHeader from '../text/sub-header/TextSubHeader';
import Label from '../form/label/Label';
import Button from '../form/button/Button';

const Result = ({ data, created, onRemove }) => (
  <div className="Result">
    <div className="App-box">
      <TextSubHeader>
        { dayjs(created).format('YYYY-MM-DD (HH:mm)') }
      </TextSubHeader>
      <div className="Result-data">
        {data.map(({ label, value, icon }) => (
          <div key={label}>
            <Label>{label}</Label>
            <span>{value} {icon}</span>
          </div>
        ))}
      </div>
      <Button as="link" onClick={onRemove}>Remove</Button>
    </div>
  </div>
);

Result.propTypes = {
  data: PropTypes.array,
  created: PropTypes.object,
  onRemove: PropTypes.func,
};

Result.defaultProps = {
  data: [],
  created: dayjs(),
  onRemove: () => {},
};

export default Result;
