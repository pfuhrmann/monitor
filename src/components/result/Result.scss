@import "../../variables.scss";

.Result {
  .App-box {
    margin-right: 2em;
  }

  &-data {
    display: flex;
    padding-bottom: 2.5em;

    div {
      display: flex;
      flex-direction: column;
      margin: 0 1.5em;
      align-self: center;
    }

    svg {
      align-self: center;
    }
  }
}
