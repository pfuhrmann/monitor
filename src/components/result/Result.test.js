import React from 'react';
import ReactDOM from 'react-dom';
import Result from './Result';
import renderer from 'react-test-renderer';
import dayjs from 'dayjs';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faSortUp } from '@fortawesome/free-solid-svg-icons';

library.add(faSortUp);

const el = <Result id={1} data={[
  { value: 'val', label: 'val', icon: null },
  { value: 'val2', label: 'val2', icon: <FontAwesomeIcon icon="sort-up" /> },
]} created={dayjs('2018-08-08')} />

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(el, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders component correctly', () => {
  const component = renderer.create(el);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
