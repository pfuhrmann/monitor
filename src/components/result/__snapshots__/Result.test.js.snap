// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div
  className="Result"
>
  <div
    className="App-box"
  >
    <h2
      className=""
    >
      2018-08-08 (00:00)
    </h2>
    <div
      className="Result-data"
    >
      <div>
        <label>
          val
        </label>
        <span>
          val
           
        </span>
      </div>
      <div>
        <label>
          val2
        </label>
        <span>
          val2
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort-up fa-w-10 "
            data-icon="sort-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
    <button
      className="Form-btn-link"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
